python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -512,7 +493,42 @@ def resolve_tags(self, _graphene_info: ResolveInfo):
         ]
 
     def resolve_runStatus(self, _graphene_info: ResolveInfo) -> GrapheneRunStatus:
-        return GrapheneBulkActionStatus(self.status).to_dagster_run_status()
+        converted_status = BulkActionStatus[self.status]
+        if converted_status is BulkActionStatus.FAILED:
+            return GrapheneRunStatus.FAILURE
+        if converted_status is BulkActionStatus.CANCELED:
+            return GrapheneRunStatus.CANCELED
+        if converted_status is BulkActionStatus.CANCELING:
+            return GrapheneRunStatus.CANCELING
+        if converted_status is BulkActionStatus.REQUESTED:
+            # if no runs have been launched:
+            if len(self._get_records(_graphene_info)) == 0:
+                return GrapheneRunStatus.NOT_STARTED  # GrapheneRunStatus.QUEUED?
+            return GrapheneRunStatus.STARTED
+        # BulkActionStatus.COMPLETED
+        sub_runs = self._get_records(_graphene_info)
+        sub_run_statuses = [record.dagster_run.status.value for record in sub_runs]
+        if all(status == "SUCCESS" for status in sub_run_statuses):
+            return GrapheneRunStatus.SUCCESS
+        if any(status == "FAILURE" for status in sub_run_statuses):
+            return GrapheneRunStatus.FAILURE
+        if any(status == "CANCELED" for status in sub_run_statuses):
+            return GrapheneRunStatus.FAILURE
+
+        # can't import this because two deserializers get registered for PipelineRunStatsSnapshot
```

Comment:
I'd rather use the code below to check the statuses of the runs, but if I import `DagsterRunStatus` I get the error `dagster._serdes.errors.SerdesUsageError: Multiple deserializers registered for storage name 'PipelineRunStatsSnapshot'`  I think because the  storage_name` for `DagsterRunStatsSnapshot`

```python
@whitelist_for_serdes(storage_name="PipelineRunStatsSnapshot")
class DagsterRunStatsSnapshot(
...
```

collides with `GraphenePipelineRunStatsSnapshot`

Not sure what there is to de about this other than move `DagsterRunStatus` into a separate module

Reply:
> Not sure what there is to de about this other than move DagsterRunStatus into a separate module

Surprised that that would help with the serdes error, but moving it to a separate module seems like a nice thing to me.